### ISSUE TITLE
Make accessibility help generic

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#183363]
Currently we have a few different accessibility help dialogs that appear, for example:

Editor:

![The editor help is a popup with a header and div](https://github.com/microsoft/vscode/assets/2193314/849ddb71-d252-4e48-9af0-3696c5fa81fd)

Terminal:

![The terminal has a monaco editor appear inline inside the terminal](https://github.com/microsoft/vscode/assets/2193314/18d544ef-a486-4253-a2a1-2615e90f24c7)

Chat view:

![The chat view reuses the prompt monaco editor](https://github.com/microsoft/vscode/assets/2193314/8db58526-b4df-4872-b582-536cf22ad121)

They each work similar but have different implementations and different commands (they probably share some code, I'm not sure).

My proposal is to have a single command "Show Accessibility Help" bound to alt+f1 across the whole workbench, components can register their own custom text which ends up showing in a centered monaco editor. From there we can trap the focus on the editor and escape or clicking out of the dialog would hide it and return focus to the previous `activeElement`.

The presentation of it could look similar to the editor dialog but have a monaco editor inside instead. It could also have a 50% black background to give a strong indication that you can click here to get out. The actual visual presentation isn't too important but with a single implementation we can make it look good just one time.

This would lead to a consistent experience across all components that use it, shared implementation, a single _lazy loaded_ monaco editor that's only constructed when needed and it wouldn't lead to potential issues such as getting the content and the help message mixed up when it's currently sharing an editor.

cc @jooyoungseo

**Comment by meganrogge** [comment]
Will address https://github.com/microsoft/vscode/issues/184176

**Comment by Tyriar** [comment]
@meganrogge should this be reopened to track all the follow ups in #184435?

**Comment by meganrogge** [comment]
Remaining help menu adoption (more complex so in separate PRs):
- notebook https://github.com/microsoft/vscode/issues/184176 
- editor https://github.com/microsoft/vscode/issues/179689
- diff editor - after that refactor has happened

To do later: 
- adopt in hovers https://github.com/microsoft/vscode/issues/173911#issuecomment-1505602012
- chat responses https://github.com/microsoft/vscode-copilot/issues/284
- notebook output cells https://github.com/microsoft/vscode/issues/183208
- improve inline chat menu https://github.com/microsoft/vscode-copilot/issues/313
- https://github.com/microsoft/vscode/issues/184612
- consider using this as the basis for terminal accessible buffers
- https://github.com/microsoft/vscode/issues/184618

**Comment by meganrogge** [comment]
All issues addressed besides the ones that will be handled by @amunger and @hediet
